### PR TITLE
Add with-google-analytics example

### PR DESCRIPTION
**Pull request by prichodko**
Adds an example showing how to setup global site tag - `gtag.js` to send event data to Google Analytics. 

`gtag.js` replaces older `analytics.js` solution and is currently recommended by Google - https://developers.google.com/analytics/devguides/collection/gtagjs/migration

Related: https://github.com/zeit/next.js/issues/160

### CODE REVIEW

**Comment by Vadorequest** [comment]
@prichodko Beware to apply this:

```js
  GA.set({page: pageName})
  GA.pageview(pageName)
```

I've read somewhere that doing a simple `pageview` isn't good, since it put the GA in some kind of broken state because it doesn't know that the page has changed if you don't use `set`. (and messes up other calls by providing the wrong page)

**Comment by prichodko** [comment]
@Vadorequest thanks for checking out and yes, this is true for old `analytics.js` - https://developers.google.com/analytics/devguides/collection/analyticsjs/single-page-applications, but is different with the new `gtag.js` - https://developers.google.com/analytics/devguides/collection/gtagjs/single-page-applications.

**Comment by mattiascibien** [comment]
@prichodko I think you should format the code using `standard` for the travis-ci build to pass and the therefore the code-coverage.

**Comment by prichodko** [comment]
@mattiascibien thanks! Haven't noticed it. :)

PR updated.

**Comment by mattiascibien** [comment]
@prichodko in any case I have included the code you provided on my site and it worked like a charm (even with static export). I will do something similar for azure appinsights I guess but I am still experimenting.

**Comment by prichodko** [comment]
@timneutkens rebased :)

File: examples/with-google-analytics/pages/_document.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import Document, { Head, Main, NextScript } from 'next/document'
+
+import { GA_TRACKING_ID } from '../lib/gtag'
+
+export default class extends Document {
+  static getInitialProps ({ renderPage }) {
+    return renderPage()
```
Comment (timneutkens) [inline examples/with-google-analytics/pages/_document.js]: const { html, head, errorHtml, chunks } = renderPage()
return { html, head, errorHtml, chunks, styles }

lets do it like in the readme

Comment (prichodko) [reply examples/with-google-analytics/pages/_document.js]: Makes sense. Done ✅.

**Comment by prichodko** [comment]
@timneutkens rebased

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Nice!

**Comment by kachkaev** [comment]
Thanks for this example @prichodko! It'd be great if `GA_TRACKING_ID` was coming from process.env – that way it's easier to configure dev vs production.

**Comment by prichodko** [comment]
@kachkaev you are definitely right! I tried to keep example as simple as possible. However for the env variables I would currently use a new feature released in `5.1` - https://zeit.co/blog/next5-1#environment-configuration. What do you think?

**Pull request merged**